src/backup.ts

```diff
@@ -0,0 +1,387 @@
+        },
+      }
+    );
+    return this.request<CreateBackupGaxOperation>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'createBackup',
+        reqOpts,
+      },
+      (err, resp) => {
+        if (err) {
+          callback!(err, null, null);
+          return;
+        }
+        callback!(null, this, resp, resp || undefined);
+      }
+    );
+  }
+
+  getMetadata(): Promise<GetMetadataResponse>;
+  getMetadata(callback: GetMetadataCallback): void;
+  /**
+   * @typedef {array} GetMetadataResponse
+   * @property {object} 0 The {@link Backup} metadata.
+   * @property {object} 1 The full API response.
+   */
+  /**
+   * @callback GetMetadataCallback
+   * @param {?Error} err Request error, if any.
+   * @param {object} metadata The {@link Backup} metadata.
+   * @param {object} apiResponse The full API response.
+   */
+  /**
+   * Retrieves backup's metadata.
+   *
+   * @see {@link #getState}
+   * @see {@link #getExpireTime}
+   *
+   * @method Backup#getMetadata
+   * @returns {Promise<GetMetadataResponse>}
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const database = spanner.database('my-database');
+   * const backup = instance.backup('my-backup');
+   * const [backupInfo] = await backup.getMetadata();
+   * console.log(`${backupInfo.name}: size=${backupInfo.sizeBytes}`);
+   */
+  getMetadata(
+    callback?: GetMetadataCallback
+  ): void | Promise<GetMetadataResponse> {
+    const reqOpts: databaseAdmin.spanner.admin.database.v1.IGetBackupRequest = {
+      name: this.formattedName_,
+    };
+    return this.request<IBackupTranslatedEnum>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'getBackup',
+        reqOpts,
+      },
+      (err, response) => {
+        callback!(err, response);
+      }
+    );
+  }
+
+  /**
+   * Retrieves the state of the backup.
+   *
+   * The backup state indicates if the backup has completed.
+   *
+   * @see {@link #getMetadata}
+   *
+   * @method Backup#getState
+   * @returns {Promise<EnumKey<typeof, databaseAdmin.spanner.admin.database.v1.Backup.State> | undefined>}
+   *     when resolved, contains the current state of the backup if it exists, or
+   *     undefined if the backup does not exist.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const myBackup = instance.backup('my-backup');
+   * const state = await myBackup.getState();
+   * const backupCompleted = (state === 'READY');
+   */
+  async getState(): Promise<
+    | EnumKey<typeof databaseAdmin.spanner.admin.database.v1.Backup.State>
+    | undefined
+  > {
+    const [backupInfo] = await this.getMetadata();
+    const state = backupInfo.state;
+    return state === null || state === undefined ? undefined : state;
```

Comment:
Would an alternative be `return backupInfo.state || undefined`?

Reply:
That's much nicer, thanks.